packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php

```diff
@@ -145,6 +146,19 @@ public function processNodes(
                 $node->expr->setAttribute(AttributeKey::SCOPE, $mutatingScope);
             }
 
+            $expr = $node;
```

Comment:
this may need to be cloned to avoid object changed, ref https://3v4l.org/BU1O0

Reply:
I tried debug after while, and it seems the object is followed, so it actually ok, see https://3v4l.org/0WYqc